plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -205,8 +206,14 @@ public void storeMergedRows(Page page)
     {
         DeltaLakeMergePage mergePage = createPages(page, tableColumnCount);
 
-        mergePage.insertionsPage().ifPresent(insertPageSink::appendPage);
-        mergePage.updateInsertionsPage().ifPresent(insertPageSink::appendPage);
+        mergePage.insertionsPage().ifPresent(insertions -> {
+            insertPageSink.appendPage(insertions);
+            writtenBytes += insertions.getSizeInBytes();
+        });
+        mergePage.updateInsertionsPage().ifPresent(updateInsertions -> {
+            insertPageSink.appendPage(updateInsertions);
+            writtenBytes += updateInsertions.getSizeInBytes();
+        });
 
         processInsertions(mergePage.insertionsPage(), INSERT_CDF_LABEL);
         processInsertions(mergePage.updateInsertionsPage(), UPDATE_POSTIMAGE_CDF_LABEL);
```

Comment:
after insert completed bytes, we should have
`writtenBytes += cdfPageSink == null ? 0 : cdfPageSink.getCompletedBytes()`